### ISSUE TITLE
Backup and restore pg_basebackup in k8s cluster

### ISSUE THREAD

**Issue by haroonb** [zalando/patroni#1507]
Hello I'm using the postgres operator to create a cluster. I have a k8s cronjob which makes a daily pg_basebackup of the cluster.

I was testing, if I could restore one of the backups, but I can"t seem to figure out how to do it and as I can't find a guide for it I was hopeing for some assistance here.

I do the following steps to restore the backup into another cluster
1. scale statefulset to replica=1
2. stop patroni pgqd cron services in the pod
3. copy backup into the pod and chown for postgres
4. start patroni pgqd cron services

I would assume it would create recreate the database as was at the time of the backup, but I get the errors like 

`acid-postgres-cluster-0 postgres 2020-04-24 10:15:39,031 CRITICAL: system ID mismatch, node acid-postgres-cluster-0 belongs to a different cluster: 6819218666252070970 != 6818158048719212611`

I tried to `patronictl failover` to force a leader, but I get 

```
root@acid-postgres-cluster-0:/home/postgres# patronictl failover
Candidate ['acid-postgres-cluster-0'] []: acid-postgres-cluster-0
Current cluster topology
+-----------------------+-------------------------+--------------+------+---------+----+-----------+
|        Cluster        |          Member         |     Host     | Role |  State  | TL | Lag in MB |
+-----------------------+-------------------------+--------------+------+---------+----+-----------+
| acid-postgres-cluster | acid-postgres-cluster-0 | 10.244.3.160 |      | stopped |    |   unknown |
+-----------------------+-------------------------+--------------+------+---------+----+-----------+
Are you sure you want to failover cluster acid-postgres-cluster? [y/N]: y
2020-04-24 10:34:31,116 - ERROR - None
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/patroni/ctl.py", line 677, in _do_failover_or_switchover
    r = request_patroni(member, 'post', action, failover_value)
  File "/usr/local/lib/python3.6/dist-packages/patroni/ctl.py", line 133, in request_patroni
    return request_executor(member, method, endpoint, data)
  File "/usr/local/lib/python3.6/dist-packages/patroni/request.py", line 52, in __call__
    url = urlunparse((scheme, netloc, endpoint, '', '', ''))
  File "/usr/lib/python3.6/urllib/parse.py", line 476, in urlunparse
    _coerce_args(*components))
  File "/usr/lib/python3.6/urllib/parse.py", line 120, in _coerce_args
    raise TypeError("Cannot mix str and non-str arguments")
TypeError: Cannot mix str and non-str arguments
2020-04-24 10:34:31,117 - WARNING - Failing over to DCS
2020-04-24 10:34:31.11788 Could not failover using Patroni api, falling back to DCS
+-----------------------+-------------------------+--------------+------+---------+----+-----------+
|        Cluster        |          Member         |     Host     | Role |  State  | TL | Lag in MB |
+-----------------------+-------------------------+--------------+------+---------+----+-----------+
| acid-postgres-cluster | acid-postgres-cluster-0 | 10.244.3.160 |      | stopped |    |   unknown |
+-----------------------+-------------------------+--------------+------+---------+----+-----------+

```

I tried `patronictl reinit`, but get 

```
root@acid-postgres-cluster-0:/home/postgres# patronictl reinit acid-postgres-cluster
+-----------------------+-------------------------+--------------+------+---------+----+-----------+
|        Cluster        |          Member         |     Host     | Role |  State  | TL | Lag in MB |
+-----------------------+-------------------------+--------------+------+---------+----+-----------+
| acid-postgres-cluster | acid-postgres-cluster-0 | 10.244.3.160 |      | stopped |    |   unknown |
+-----------------------+-------------------------+--------------+------+---------+----+-----------+
Which member do you want to reinitialize [acid-postgres-cluster-0]? []: acid-postgres-cluster-0
Are you sure you want to reinitialize members acid-postgres-cluster-0? [y/N]: y
Failed: reinitialize for member acid-postgres-cluster-0, status code=503, (Cluster has no leader, can not reinitialize)
```

I don't really understand what I'm doing wrong or missing. 

Thanks for any assistance

**Comment by CyberDem0n** [comment]
> I do the following steps to restore the backup into another cluster
scale statefulset to replica=1
stop patroni pgqd cron services in the pod
copy backup into the pod and chown for postgres

It is fundamentally not possible to restore the binary backup into another cluster. Every cluster has the unique identifier and Patroni stores it in the DCS and performs a check which leads to the following error:
> acid-postgres-cluster-0 postgres 2020-04-24 10:15:39,031 CRITICAL: system ID mismatch, node acid-postgres-cluster-0 belongs to a different cluster: 6819218666252070970 != 6818158048719212611

The proper way of restoring backups - deploying the new cluster which is configured to perform the backup restore, aka [custom bootstrap](https://patroni.readthedocs.io/en/latest/replica_bootstrap.html) instead of initdb.

In the Operator and Spilo it is known as ["clone"](https://github.com/zalando/postgres-operator/blob/master/docs/user.md#how-to-clone-an-existing-postgresql-cluster).

After going through all layers, the clone configuration is turned into the custom bootstrap configuration in the postgres.yml.

**Comment by haroonb** [comment]
> It is fundamentally not possible to restore the binary backup into another cluster.

Then Patroni works differently from the bitnami we used before, there we could just terminate the pod before it startet the database, mount the pv to a busybox and copy alle the files over and start the postgre pod.

> The proper way of restoring backups - deploying the new cluster which is configured to perform the backup restore, aka custom bootstrap instead of initdb.

I have read the documentation you linked, but I don't grasp how I could apply this to my current situation. I'm currently only left with the pg_basebackup binary files.

> After going through all layers, the clone configuration is turned into the custom bootstrap configuration in the postgres.yml

Are you referring to this [yaml-file](https://github.com/zalando/postgres-operator/blob/master/manifests/complete-postgres-manifest.yaml)?
I would assume, all the bootstrap configuration is to be placed under `patroni:` and `initdb:` part to be removed?

I'm really trying to understand the documentation, but I find it hard to follow, that maybe to the fact that I only started to work with postgres and patroni. Is there a more detailed documentation with examples which I could read into.

Thank you for the response and assistance.
